### ISSUE TITLE
Allow specifying type for `NSObject.reactive.signal`

### ISSUE THREAD

**Issue by Zeeker** [ReactiveCocoa/ReactiveCocoa#3451]
With the deprecation of `values(forKeyPath:)` and addition of `signal(forKeyPath:)` and `producer(forKeyPath:)`, I think it would be nice to make use of Swifts generics to specify the type of the `signal`/`producer`.

Using `filterMap` this can already be achieved, but a ready-made solution in ReactiveCocoa would be great.

### Example

```swift
func signal<Value>(forKeyPath: String, withValueType: Value.Type) -> Signal<Value, NoError>
```

**Comment by andersio** [comment]
This could wait until Smart Key Path in Swift is implemented, since it would likely come with KVC interoperability for ObjC classes according to the WIP PR.

**Comment by andersio** [comment]
Resolved in #3491 with Smart Key Path. :tada:
